base/shell/explorer/traywnd.cpp

```diff
@@ -2946,59 +2717,111 @@ class CTrayWindow :
+                if (prcStartBtn->right + (int)pwi->cxWindowBorders * 2 + 1 < pwi->rcWindow.right &&
+                    ppt->x > prcStartBtn->right)
                 {
-                    return 0;
+                    return FALSE;
                 }
                 break;
             }
             case ABE_BOTTOM:
             {
-                if (pt.x > rcStartBtn.right || pt.y < rcStartBtn.top)
-                    return 0;
+                if (ppt->x > prcStartBtn->right || ppt->y < prcStartBtn->top)
+                    return FALSE;
 
-                if (rcStartBtn.bottom + (int)wi.cyWindowBorders * 2 + 1 < wi.rcWindow.bottom &&
-                    pt.y > rcStartBtn.bottom)
+                if (prcStartBtn->bottom + (int)pwi->cyWindowBorders * 2 + 1 < pwi->rcWindow.bottom &&
+                    ppt->y > prcStartBtn->bottom)
                 {
-                    return 0;
+                    return FALSE;
                 }
 
                 break;
             }
         }
+        return TRUE;
+    }
+
+    BOOL IsPointWithinShowDesktopButton(LPPOINT ppt, LPRECT prcShowDesktopBtn, WINDOWINFO* pwi)
+    {
+        if (!ppt || !prcShowDesktopBtn)
+            return FALSE;
+        UNREFERENCED_PARAMETER(pwi);
+        /*if (!pwi)
+            return FALSE;*/
+        //InflateRect(prcShowDesktopBtn, (int)pwi->cxWindowBorders * 2 + 1, (int)pwi->cyWindowBorders * 2 + 1);
+
+        switch (m_Position)
+        {
+            case ABE_LEFT:
+            {
+                if (ppt->x > prcShowDesktopBtn->right || ppt->y < prcShowDesktopBtn->top)
+                    return FALSE;
+                break;
+            }
+            case ABE_TOP:
+            {
+                if (ppt->x < prcShowDesktopBtn->left || ppt->y > prcShowDesktopBtn->bottom)
+                    return FALSE;
+                break;
+            }
+            case ABE_RIGHT:
+            {
+                if (ppt->x < prcShowDesktopBtn->left || ppt->y < prcShowDesktopBtn->top)
+                    return FALSE;
+                break;
+            }
+            case ABE_BOTTOM:
+            {
+                if (ppt->x < prcShowDesktopBtn->left || ppt->y < prcShowDesktopBtn->top)
+                    return FALSE;
+                break;
+            }
+        }
+        return TRUE;
+    }
+
+    LRESULT OnNcLButtonDown(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+    {
+        /* This handler implements the trick that makes  the start button to
+           get pressed when the user clicked left or below the button */
+
+        POINT pt = {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+        WINDOWINFO wi = {sizeof(WINDOWINFO)};
+
+        bHandled = FALSE;
+
+        RECT rcStartBtn;
+        m_StartButton.GetWindowRect(&rcStartBtn);
+        GetWindowInfo(m_hWnd, &wi);
+
+        if (IsPointWithinStartButton(&pt, &rcStartBtn, &wi))
+        {
+            bHandled = TRUE;
+            PopupStartMenu();
```

Comment:
If this is executed, should the code below be run, or should there be a `return` in this?

Reply:
Actually yeah, I guess there should...